packages/@react-aria/numberfield/docs/useNumberField.mdx

```diff
@@ -0,0 +1,293 @@
+
+## Example
+
+```tsx example export=true
+import {useNumberFieldState} from '@react-stately/numberfield';
+import {useLocale} from '@react-aria/i18n';
+import {useButton} from '@react-aria/button';
+
+function NumberField(props) {
+  let {locale} = useLocale();
+  let state = useNumberFieldState({...props, locale});
+  let inputRef = React.useRef();
+  let {
+    labelProps,
+    numberFieldProps,
+    inputFieldProps,
+    incrementButtonProps,
+    decrementButtonProps
+  } = useNumberField({...props, inputRef}, state);
+
+  let {buttonProps: incrementProps} = useButton(incrementButtonProps);
+  let {buttonProps: decrementProps} = useButton(decrementButtonProps);
+
+  return (
+    <div>
+      <label {...labelProps}>{props.label}</label>
+      <div {...numberFieldProps}>
+        <button {...decrementProps}>-</button>
+        <input {...inputFieldProps} ref={inputRef} />
+        <button {...incrementProps}>+</button>
+      </div>
+    </div>
+  );
+}
+
+<NumberField
+  label="Price"
+  defaultValue={6}
+  formatOptions={{
+    style: 'currency',
+    currency: 'USD'
+  }} />
+```
+
+## Usage
+
+The following examples show how to use the `NumberField` component created in the above example.
+
+### Controlled
+
+By default, `NumberField` is uncontrolled. However, when using the `value` prop, it becomes controlled.
+This allows you to store the current value in your own state, and use it elsewhere.
+
+The `onChange` event is triggered whenever the number value updates. This happens when the user types a
+value and blurs the input, or when incrementing or decrementing the value. It does not happen as the user
+types because partial input may not be parseable to a valid number.
+
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState(6);
+
+  return (
+    <>
+      <NumberField
+        label="Controlled value"
+        value={value}
+        onChange={setValue} />
+      <div>Current value prop: {value}</div>
+    </>
+  );
+}
+```
+
+### Decimals
+
+The default formatting style for `NumberField` is decimal, but you can configure various aspects via the `formatOptions`
+prop. All options supported by [Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat)
+are supported, including configuration of minimum and maximum fraction digits, sign display, grouping separators, etc.
+
+The following example uses the `signDisplay` option to include the plus sign for positive numbers, for example to display
+an offset from some value. In addition, it always displays a minimum of 1 digit after the decimal point, and allows up to
+2 fraction digits. If the user enters more than 2 fraction digits, the result will be rounded.
+
+```tsx example
+<NumberField
+  label="Adjust exposure"
+  defaultValue={0}
+  formatOptions={{
+    signDisplay: 'exceptZero',
+    minimumFractionDigits: 1,
+    maximumFractionDigits: 2
+  }} />
+```
+
+### Percentages
```

Comment:
Was just thinking about this, is it odd that percent just works for NumberField, but for Slider you have to do extra work? Should slider be updated to do the same while it's still alpha?
https://reactspectrum.blob.core.windows.net/reactspectrum/f2ee85f699ea74eb6c402905f9db9f775477b797/docs/react-aria/useNumberField.html#percentages vs https://react-spectrum.adobe.com/react-aria/useSlider.html#examples

Also, should we add a section to the slider docs to talk about percentage specifically? All of these sections seem useful for Slider docs

Reply:
slider isn't in alpha... I'm not sure what you're asking though

Reply:
ah, yep, brain fart, it's not in alpha anymore, but what I'm asking is for slider, we do this
```
<Slider
  label="Opacity"
  formatOptions={{style: 'percent'}}
  maxValue={1}
  step={0.01}
/>
```

but for numberfield we do this
```
<NumberField
  label="Sales tax"
  defaultValue={0.5}
  formatOptions={{style: 'percent'}}
/>
```

Note that in one we have to set the step and in the other, we don't

Reply:
discussed offline, we're willing to have numberfield differ from slider in this matter since slider is already released and we like the feature, so don't want to omit from numberfield